### PR TITLE
when installing init to root dir, skip init symlink

### PR DESCRIPTION
**Pull request by mpictor**

### CODE REVIEW

File: pkg/uroot/uroot.go [hunk 1]
```diff
@@ -255,7 +255,9 @@ func CreateInitramfs(logger logger.Logger, opts Opts) error {
 			if err != nil {
 				return err
 			}
-			if err := archive.AddRecord(cpio.Symlink("init", rtarget)); err != nil {
+			if rtarget == "init" {
+				logger.Printf("not creating symlink init -> init")
```
Comment (hugelgupf) [inline pkg/uroot/uroot.go]: Capitalize Not, and say why

Not creating symlink init -> init, most likely because init already exists

**Comment by hugelgupf** [comment]
Can you add a test to the top-level uroot_test for your use case?

**Comment by rminnich** [comment]
I still want to hear why this PR is necessary given that I think I showed it is not?

File: pkg/uroot/uroot.go (hunk 1 above)
Comment (rminnich) [reply pkg/uroot/uroot.go]: will this be another warning people ignore to their later surprise? Should it be an error with some explanation? 
Sorry for my confusion on this one.

Comment (rminnich) [reply pkg/uroot/uroot.go]: OK, how about this:
return fmt.Errorf("You are replacing u-root /init with your own. If this is your intent, you must also specify -initcmd=\"\" on the command line")

I realize it's a bit verbose but the consequences of getting this wrong, especially in firmware, are pretty dire.

**Comment by mpictor** [comment]
> I still want to hear why this PR is necessary given that I think I showed it is not?

Sure, I'll see if it's possible. 

Note that I'm doing it programmatically, not from the command line... kinda thinking I already have the suggested args.

File: pkg/uroot/uroot.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/uroot/uroot.go]: @mpictor I'd be good with this

**Comment by mpictor** [comment]
I added a test, which passes with or without my change. After doing some digging, it seems my problem only exists when using the `vmtest` package - in `QEMU()` it has
```go
		if len(o.BuildOpts.InitCmd) == 0 {
			o.BuildOpts.InitCmd = "init"
		}
```
... so the question is, should I modify that package? I'm not sure this is a problem anyone else is likely to run into, so it might make more sense for me change my code.

**Comment by mpictor** [comment]
Thinking it makes more sense for me to modify my code. Closing.

**Pull request closed without merging**